### PR TITLE
Don't roll on upgrades from 3.0

### PR DESCRIPTION
**Pull request by russgold**
Takes into consideration that in 3.0, the prometheus annotations were computed differently than in 3.1.

### CODE REVIEW

**Review by rjeberhard — APPROVED** [review]

**Review by jshum2479 — COMMENTED** [review]
LGTM,  the only comment I have is when a pod needs to be replaced because of domain zip hash changed, the operator logs a message "Replacing pod ... because of null"  - is there a way to put the actual reason there?

**Comment by rjeberhard** [comment]
> LGTM, the only comment I have is when a pod needs to be replaced because of domain zip hash changed, the operator logs a message "Replacing pod ... because of null" - is there a way to put the actual reason there?

I'll file a bug, but I don't want to hold up an initial release. We've implemented a series of compatibility checks, which must not be complete. If none of them match then the result is "null".

**Comment by russgold** [comment]
> > LGTM, the only comment I have is when a pod needs to be replaced because of domain zip hash changed, the operator logs a message "Replacing pod ... because of null" - is there a way to put the actual reason there?
> 
> I'll file a bug, but I don't want to hold up an initial release. We've implemented a series of compatibility checks, which must not be complete. If none of them match then the result is "null".

Then do file a bug and we can deal with it in the next release. I have no test for this and did not notice it.

**Pull request merged**
